File: scratch/extract_mapping.py
import requests
import re

url = "https://chomered.com/book/chapter/18690052"
response = requests.get(url)
html = response.text

# Find CSS files
css_files = re.findall(r'href="(https?://[^"]+\.css)"', html)
print("CSS Files found:")
for css in css_files:
    print(css)
    if "novel" in css or "style" in css:
        css_content = requests.get(css).text
        if ".icon-" in css_content:
            print(f"Mapping found in {css}")
            mappings = re.findall(r'\.icon-(\d+):before\{content:"([^"]+)"\}', css_content)
            for m in mappings:
                print(f"{m[0]}: {m[1]}")

File: scratch/extract_mapping.js
async function extract() {
    const url = "https://chomered.com/book/chapter/18690052";
    try {
        const response = await fetch(url);
        const html = await response.text();

        // Find CSS files (handle relative URLs too)
        const cssFiles = html.match(/href="([^"]+\.css[^"]*)"/g) || [];
        console.log("CSS Files found:");
        for (let cssLink of cssFiles) {
            let cssUrl = cssLink.match(/href="([^"]+)"/)[1];
            if (!cssUrl.startsWith('http')) {
                cssUrl = new URL(cssUrl, url).href;
            }
            console.log(cssUrl);
            try {
                const cssResponse = await fetch(cssUrl);
                const cssContent = await cssResponse.text();
                if (cssContent.includes(".icon-")) {
                    console.log(`Mapping found in ${cssUrl}`);
                    const mappings = cssContent.match(/\.icon-(\d+):before\s*\{\s*content\s*:\s*"([^"]+)"\s*\}/g) || [];
                    const mappingObj = {};
                    mappings.forEach(m => {
                        const parts = m.match(/\.icon-(\d+):before\s*\{\s*content\s*:\s*"([^"]+)"\s*\}/);
                        if (parts) {
                            mappingObj[parts[1]] = parts[2];
                        }
                    });
                    console.log(JSON.stringify(mappingObj, null, 2));
                    return; // Stop after finding the first one with icons
                }
            } catch (e) {
                console.error(`Failed to fetch ${cssUrl}`);
            }
        }
    } catch (error) {
        console.error(error);
    }
}

extract();
